src/cli/bin.js

```diff
@@ -23,8 +23,14 @@ const debug = require('debug')('ipfs:cli')
 const pkg = require('../../package.json')
 const parser = require('./parser')
 const commandAlias = require('./command-alias')
+const semver = require('semver')
 
 async function main (args) {
+  if (!semver.gt(process.versions.node, semver.minVersion(pkg.engines.node))) {
```

Comment:
Should this be `semver.gte(...)`?

Reply:
How about using `satisfies` instead to simplify?

```suggestion
  if (!semver.satisfies(process.versions.node, pkg.engines.node)) {
```

Reply:
This check needs to be the first thing we do in this file (after requiring `semver` and the `package.json`) because any one of the other requires could contain syntax that is incompatible with the node version we're compatible with.